backtester/engine/setup.go

```diff
@@ -216,60 +204,110 @@ func NewFromConfig(cfg *config.Config, templatePath, output string) (*BackTest,
 			return nil, fmt.Errorf("could not format currency %v, %w", curr, err)
 		}
 		curr = curr.Format(requestFormat.Delimiter, requestFormat.Uppercase)
-		err = exchBase.CurrencyPairs.EnablePair(a, curr)
-		if err != nil && !errors.Is(err, currency.ErrPairAlreadyEnabled) {
-			return nil, fmt.Errorf(
-				"could not enable currency %v %v %v. Err %w",
-				cfg.CurrencySettings[i].ExchangeName,
-				cfg.CurrencySettings[i].Asset,
-				cfg.CurrencySettings[i].Base+cfg.CurrencySettings[i].Quote,
-				err)
+		var avail, enabled currency.Pairs
+		avail, err = exch.GetAvailablePairs(a)
+		if err != nil {
+			return nil, fmt.Errorf("could not format currency %v, %w", curr, err)
+		}
+		enabled, err = exch.GetEnabledPairs(a)
+		if err != nil {
+			return nil, fmt.Errorf("could not format currency %v, %w", curr, err)
+		}
+
+		avail = avail.Add(curr)
```

Comment:
The `Add` function already does that. This will ensure that the supplied currency is available for use, whether its from DB, CSV, as in, you can use whatever currency name you'd like. If using API data, if it doesn't exist then it will fail at that stage anyway:
`Could not setup backtester from config. Error: could not retrieve candle data for FTX futures BTC.OL-20210924, FTX unsuccessful HTTP status code: 404 raw response: {"success":false,"error":"No such market: BTC.OL-20210924 "}`

Reply:
Ahh right okay! Thanks for the explanation 👍